haystack/document_stores/pinecone.py

```diff
@@ -1451,7 +1451,7 @@ def _label_to_meta(self, labels: list) -> dict:
                         "label-answer-type": label.answer.type,
                         "label-answer-score": label.answer.score,
                         "label-answer-context": label.answer.context,
-                        "label-answer-document-id": label.answer.document_id,
+                        "label-answer-document-ids": label.answer.document_ids,
```

Comment:
PineconeDocumentStore cannot store labels, but casts them into documents by storing its values inside `Document.meta`. We just apply the changes for `document_ids` here.